# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1524666202 14400
#      Wed Apr 25 10:23:22 2018 -0400
# Branch stable
# Node ID 0b253ba6eb91adb8980b1619dd0956310fd2e725
# Parent  7d32780f4dea1c5d7d2b4eb25e8a365ee059b227
avoid syntax errors in build-aux/subst scripts

* m4/acinclude.m4: Use ${top_srcdir} instead of $(top_srcdir) to avoid
syntax errors in shell scripts.

* subst-config-vals.in.sh, subst-cross-config-vals.in.sh:
Also define srcdir, top_srcdir, abs_srcdir, and abs_top_srcdir.
Define directory variables ahead of any other variables that might use
them.

diff --git a/build-aux/subst-config-vals.in.sh b/build-aux/subst-config-vals.in.sh
--- a/build-aux/subst-config-vals.in.sh
+++ b/build-aux/subst-config-vals.in.sh
@@ -15,16 +15,31 @@
 # GNU General Public License for more details.
 #
 # You should have received a copy of the GNU General Public License
 # along with Octave; see the file COPYING.  If not, see
 # <https://www.gnu.org/licenses/>.
 
 : ${SED=@SED@}
 
+prefix="@prefix@"
+exec_prefix="@exec_prefix@"
+
+bindir=`echo "@bindir@" | sed "s|^${exec_prefix}/||"`
+libdir=`echo "@libdir@" | sed "s|^${exec_prefix}/||"`
+octlibdir=`echo "@octlibdir@" | sed "s|^${exec_prefix}/||"`
+
+includedir=`echo "@includedir@" | sed "s|^${prefix}/||"`
+octincludedir=`echo "@octincludedir@" | sed "s|^${prefix}/||"`
+
+srcdir="@srcdir@"
+top_srcdir="@top_srcdir@"
+abs_srcdir="@abs_srcdir@"
+abs_top_srcdir="@abs_top_srcdir@"
+
 NO_OCT_FILE_STRIP="@NO_OCT_FILE_STRIP@"
 AMD_CPPFLAGS="@AMD_CPPFLAGS@"
 AMD_LDFLAGS="@AMD_LDFLAGS@"
 AMD_LIBS="@AMD_LIBS@"
 AR="@AR@"
 ARFLAGS="@ARFLAGS@"
 ARPACK_CPPFLAGS="@ARPACK_CPPFLAGS@"
 ARPACK_LDFLAGS="@ARPACK_LDFLAGS@"
@@ -173,26 +188,16 @@ XTRA_CFLAGS="@XTRA_CFLAGS@"
 XTRA_CXXFLAGS="@XTRA_CXXFLAGS@"
 YACC="@YACC@"
 YFLAGS="@YFLAGS@"
 Z_CPPFLAGS="@Z_CPPFLAGS@"
 Z_LDFLAGS="@Z_LDFLAGS@"
 Z_LIBS="@Z_LIBS@"
 config_opts="@config_opts@"
 
-prefix="@prefix@"
-exec_prefix="@exec_prefix@"
-
-bindir=`echo "@bindir@" | sed "s|^${exec_prefix}/||"`
-libdir=`echo "@libdir@" | sed "s|^${exec_prefix}/||"`
-octlibdir=`echo "@octlibdir@" | sed "s|^${exec_prefix}/||"`
-
-includedir=`echo "@includedir@" | sed "s|^${prefix}/||"`
-octincludedir=`echo "@octincludedir@" | sed "s|^${prefix}/||"`
-
 $SED \
   -e "s|%NO_EDIT_WARNING%|DO NOT EDIT!  Generated automatically by subst-config-vals.|" \
   -e "s|%NO_OCT_FILE_STRIP%|${NO_OCT_FILE_STRIP}|" \
   -e "s|%OCTAVE_BINDIR%|\"${bindir}\"|" \
   -e "s|%OCTAVE_CONF_AMD_CPPFLAGS%|\"${AMD_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_AMD_LDFLAGS%|\"${AMD_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_AMD_LIBS%|\"${AMD_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_AR%|\"${AR}\"|" \
diff --git a/build-aux/subst-cross-config-vals.in.sh b/build-aux/subst-cross-config-vals.in.sh
--- a/build-aux/subst-cross-config-vals.in.sh
+++ b/build-aux/subst-cross-config-vals.in.sh
@@ -27,16 +27,31 @@
 ##   OCTAVE_CONF_MKOCTFILE_CC
 ##   OCTAVE_CONF_MKOCTFILE_CXX
 ##   OCTAVE_CONF_MKOCTFILE_DL_LD
 ##   OCTAVE_CONF_MKOCTFILE_DL_LDFLAGS
 ##   OCTAVE_CONF_MKOCTFILE_F77
 ##   OCTAVE_CONF_MKOCTFILE_LD_CXX
 ##   OCTAVE_CONF_MKOCTFILE_RANLIB
 
+prefix="@prefix@"
+exec_prefix="@exec_prefix@"
+
+bindir=`echo "@bindir@" | sed "s|^${exec_prefix}/||"`
+libdir=`echo "@libdir@" | sed "s|^${exec_prefix}/||"`
+octlibdir=`echo "@octlibdir@" | sed "s|^${exec_prefix}/||"`
+
+includedir=`echo "@includedir@" | sed "s|^${prefix}/||"`
+octincludedir=`echo "@octincludedir@" | sed "s|^${prefix}/||"`
+
+srcdir="@srcdir@"
+top_srcdir="@top_srcdir@"
+abs_srcdir="@abs_srcdir@"
+abs_top_srcdir="@abs_top_srcdir@"
+
 NO_OCT_FILE_STRIP="@NO_OCT_FILE_STRIP@"
 AMD_CPPFLAGS="@AMD_CPPFLAGS@"
 AMD_LDFLAGS="@AMD_LDFLAGS@"
 AMD_LIBS="@AMD_LIBS@"
 AR="@AR@"
 ARFLAGS="@ARFLAGS@"
 ARPACK_CPPFLAGS="@ARPACK_CPPFLAGS@"
 ARPACK_LDFLAGS="@ARPACK_LDFLAGS@"
@@ -177,26 +192,16 @@ XTRA_CFLAGS="@XTRA_CFLAGS@"
 XTRA_CXXFLAGS="@XTRA_CXXFLAGS@"
 YACC="@YACC@"
 YFLAGS="@YFLAGS@"
 Z_CPPFLAGS="@Z_CPPFLAGS@"
 Z_LDFLAGS="@Z_LDFLAGS@"
 Z_LIBS="@Z_LIBS@"
 config_opts="@config_opts@"
 
-prefix="@prefix@"
-exec_prefix="@exec_prefix@"
-
-bindir=`echo "@bindir@" | sed "s|^${exec_prefix}/||"`
-libdir=`echo "@libdir@" | sed "s|^${exec_prefix}/||"`
-octlibdir=`echo "@octlibdir@" | sed "s|^${exec_prefix}/||"`
-
-includedir=`echo "@includedir@" | sed "s|^${prefix}/||"`
-octincludedir=`echo "@octincludedir@" | sed "s|^${prefix}/||"`
-
 $SED \
   -e "s|%NO_EDIT_WARNING%|DO NOT EDIT!  Generated automatically by subst-config-vals.|" \
   -e "s|%NO_OCT_FILE_STRIP%|${NO_OCT_FILE_STRIP}|" \
   -e "s|%OCTAVE_BINDIR%|\"${bindir}\"|" \
   -e "s|%OCTAVE_CONF_AMD_CPPFLAGS%|\"${AMD_CPPFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_AMD_LDFLAGS%|\"${AMD_LDFLAGS}\"|" \
   -e "s|%OCTAVE_CONF_AMD_LIBS%|\"${AMD_LIBS}\"|" \
   -e "s|%OCTAVE_CONF_AR%|\"${AR}\"|" \
diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -2931,17 +2931,17 @@ EOF
 
 I wasn't able to find a suitable style for declaring a push-pull
 parser in a bison input file so I'm disabling bison.
 "
     OCTAVE_CONFIGURE_WARNING([warn_bison_push_pull_decl_style])
   fi
 
   if test $tmp_have_bison = no; then
-    YACC='$(top_srcdir)/build-aux/missing bison'
+    YACC='${top_srcdir}/build-aux/missing bison'
     warn_bison="
 
 I didn't find bison, or the version of bison that I found does not
 support all the features that are required, but it's only a problem
 if you need to reconstruct parse.cc, which is the case if you're
 building from VCS sources.
 "
     OCTAVE_CONFIGURE_WARNING([warn_bison])
@@ -2969,17 +2969,17 @@ AC_DEFUN([OCTAVE_PROG_FLEX], [
   AC_PROG_LEX
   case "`$LEX --version`" in
     *flex*)
       LFLAGS="-I"
       AC_MSG_RESULT([defining LFLAGS to be $LFLAGS])
       LEXLIB=
     ;;
     *)
-      LEX='$(top_srcdir)/build-aux/missing flex'
+      LEX='${top_srcdir}/build-aux/missing flex'
       warn_flex="
 
 I didn't find flex, but it's only a problem if you need to reconstruct
 lex.cc, which is the case if you're building from VCS sources.
 "
       OCTAVE_CONFIGURE_WARNING([warn_flex])
     ;;
   esac
@@ -2994,17 +2994,17 @@ AC_DEFUN([OCTAVE_PROG_GHOSTSCRIPT], [
       ac_octave_gs_names="gswin32c gs mgs"
     ;;
     *)
       ac_octave_gs_names="gs"
     ;;
   esac
   AC_CHECK_PROGS(GHOSTSCRIPT, [$ac_octave_gs_names])
   if test -z "$GHOSTSCRIPT"; then
-    GHOSTSCRIPT='$(top_srcdir)/build-aux/missing gs'
+    GHOSTSCRIPT='${top_srcdir}/build-aux/missing gs'
     warn_ghostscript="
 
 I didn't find ghostscript, so reconstructing figures for the manual
 will fail, and saving graphics in some output formats will fail when
 using Octave
 "
     OCTAVE_CONFIGURE_WARNING([warn_ghostscript])
   fi
@@ -3035,34 +3035,34 @@ Octave's plotting commands.
   AC_SUBST(GNUPLOT)
 ])
 dnl
 dnl Check for gperf.
 dnl
 AC_DEFUN([OCTAVE_PROG_GPERF], [
   AC_CHECK_PROG(GPERF, gperf, gperf, [])
   if test -z "$GPERF"; then
-    GPERF='$(top_srcdir)/build-aux/missing gperf'
+    GPERF='${top_srcdir}/build-aux/missing gperf'
     warn_gperf="
 
 I didn't find gperf, but it's only a problem if you need to
 reconstruct oct-gperf.h
 "
     OCTAVE_CONFIGURE_WARNING([warn_gperf])
-    GPERF='$(top_srcdir)/build-aux/missing gperf'
+    GPERF='${top_srcdir}/build-aux/missing gperf'
   fi
   AC_SUBST(GPERF)
 ])
 dnl
 dnl Find icotool program.
 dnl
 AC_DEFUN([OCTAVE_PROG_ICOTOOL], [
   AC_CHECK_PROG(ICOTOOL, icotool, icotool, [])
   if test -z "$ICOTOOL"; then
-    ICOTOOL='$(top_srcdir)/build-aux/missing icotool'
+    ICOTOOL='${top_srcdir}/build-aux/missing icotool'
     warn_icotool="
 
 I didn't find icotool, but it's only a problem if you need to
 reconstruct octave-logo.ico, which is the case if you're building from
 VCS sources.
 "
     OCTAVE_CONFIGURE_WARNING([warn_icotool])
   fi
@@ -3161,17 +3161,17 @@ AC_DEFUN([OCTAVE_PROG_PYTHON], [
   AC_SUBST(PYTHON)
 ])
 dnl
 dnl Find rsvg-convert program.
 dnl
 AC_DEFUN([OCTAVE_PROG_RSVG_CONVERT], [
   AC_CHECK_PROG(RSVG_CONVERT, rsvg-convert, rsvg-convert, [])
   if test -z "$RSVG_CONVERT"; then
-    RSVG_CONVERT='$(top_srcdir)/build-aux/missing rsvg-convert'
+    RSVG_CONVERT='${top_srcdir}/build-aux/missing rsvg-convert'
     warn_rsvg_convert="
 
 I didn't find rsvg-convert, but it's only a problem if you need to
 reconstruct octave-logo-*.png, which is the case if you're building
 from VCS sources.
 "
     OCTAVE_CONFIGURE_WARNING([warn_rsvg_convert])
   fi
@@ -3269,17 +3269,17 @@ dnl If we have a valid GNU tar program, 
   AC_MSG_RESULT([$REPRODUCIBLE_TAR_FLAGS])
 ])
 dnl
 dnl Check for texi2dvi.
 dnl
 AC_DEFUN([OCTAVE_PROG_TEXI2DVI], [
   AC_CHECK_PROG(TEXI2DVI, texi2dvi, texi2dvi, [])
   if test -z "$TEXI2DVI"; then
-    TEXI2DVI='$(top_srcdir)/build-aux/missing texi2dvi'
+    TEXI2DVI='${top_srcdir}/build-aux/missing texi2dvi'
     warn_texi2dvi="
 
 I didn't find texi2dvi, but it's only a problem if you need to
 reconstruct the DVI version of the manual
 "
     OCTAVE_CONFIGURE_WARNING([warn_texi2dvi])
   fi
   AC_SUBST(TEXI2DVI)
@@ -3295,17 +3295,17 @@ AC_DEFUN([OCTAVE_PROG_TEXI2PDF], [
     if test -n "$TEXI2DVI"; then
       TEXI2PDF="$TEXI2DVI --pdf"
       ac_octave_texi2pdf_missing=no;
     fi
   else
     ac_octave_texi2pdf_missing=no;
   fi
   if test $ac_octave_texi2pdf_missing = yes; then
-    TEXI2PDF='$(top_srcdir)/build-aux/missing texi2pdf'
+    TEXI2PDF='${top_srcdir}/build-aux/missing texi2pdf'
     warn_texi2pdf="
 
 I didn't find texi2pdf, but it's only a problem if you need to
 reconstruct the PDF version of the manual
 "
     OCTAVE_CONFIGURE_WARNING([warn_texi2pdf])
   fi
   AC_SUBST(TEXI2PDF)
